src/TodoContext.tsx

```diff
@@ -0,0 +1,266 @@
+
+export const TodoProvider: React.FC<TodoContextProps> = ({ children }) => {
+  const [postTodo, setPostTodo] = useState('');
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filteredTodos, setFilteredTodos] = useState<Todo[]>([]);
+  const [filter, setFilter] = useState('all');
+  const [error, setError] = useState('');
+  const [disableInput, setDisableInput] = useState(false);
+  const [loadingTodos, setLoadingTodos] = useState<number[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isChosenToRename, setIsChosenToRename] = useState(0);
+  const [editingTodo, setEditingTodo] = useState('');
+  const [handleEditing, setHandleEditing] = useState(0);
+
+  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+  const titleField = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (titleField.current) {
+      titleField.current.focus();
+    }
+  }, [tempTodo, todos.length]);
+
+  const existingCompleted = todos.some((todo) => {
+    return todo.completed;
+  });
+
+  const nonCompletedTodos = todos.reduce((counter, todo) => {
+    if (!todo.completed) {
+      return counter + 1;
+    }
+
+    return counter;
+  }, 0);
+
+  useEffect(() => {
+    api.getTodos(USER_ID)
+      .then((apiTodos) => {
+        setTodos(apiTodos);
+        setFilteredTodos(apiTodos);
+      })
+      .catch(() => {
+        setError('Unable to load todos');
+      });
+  }, []);
+
+  useEffect(() => {
+    const filteredtodo = () => {
+      switch (filter) {
+        case 'active':
+          return todos.filter((todo) => {
+            return !todo.completed;
+          });
+        case 'completed':
+          return todos.filter((todo) => {
+            return todo.completed;
+          });
+        default:
+          return todos;
+      }
+    };
+
+    setFilteredTodos(filteredtodo());
+  }, [filter, todos]);
+
+  const handleSubmit = () => {
+    if (!postTodo.trim()) {
+      setDisableInput(false);
+
+      return;
+    }
+
+    const newTodo = {
+      id: 0,
+      userId: USER_ID,
+      title: postTodo.trim(),
+      completed: false,
+    };
+
+    setTempTodo(newTodo);
+
+    api.postTodos(newTodo)
+      .then((todo) => {
+        setTempTodo(null);
+        setTodos((currentTodos) => [...currentTodos, todo]);
+        setPostTodo('');
+      })
+      .catch(() => {
+        setTempTodo(null);
+        setError('Unable to add a todo');
+      })
+      .finally(() => {
+        titleField.current?.focus();
+        setDisableInput(false);
+      });
+  };
+
+  const handleDelete = (id: number) => {
+    setLoadingTodos((currentLoading) => [...currentLoading, id]);
+
+    api.deletTodos(id)
+      .then(() => {
+        setTodos((currentTodos) => currentTodos.filter(
+          (todo) => todo.id !== id,
+        ));
+      })
+      .catch(() => {
+        setError('Unable to delete a todo');
+        if (titleField.current) {
+          titleField.current.focus();
+        }
```

Comment:
I think you can delete it, because when the todo will be deleted `todos.length` will change and   titleField.current.focus(); will work in line 68

![image](https://github.com/mate-academy/react_todo-app-with-api/assets/91946036/7317a7ed-a598-42e0-b7ae-78909e2bca09)